civismlext/preprocessing.py

```diff
@@ -157,7 +183,7 @@ def _create_col_names(self, X):
         for col in unexpanded_cnames:
             if col in self._cols_to_expand:
                 col_levels = self.levels_[col]
-                if self.dummy_na:
+                if self._dummy_na in ['expanded', 'all']:
                     # avoid exposing the sentinel to the user by replacing
                     # it with 'NaN'. If 'NaN' is already a level, use the
                     # sentinel to prevent column name duplicates.
```

Comment:
Maybe it's worth expanding this comment to indicate that if `_dummy_na` is "all", then the sentinel will only be present if there's actually a missing value. I missed / forgot that the first time I read through the code.